rollo-x1038: init at 1.8.4

ChocolateLoverRaj opened the pull request:
Tested with a Rollo X1038 printer.


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ChocolateLoverRaj commented:
Idk how to fix the restricted mode error. Can someone help?

acid-bong commented on pkgs/by-name/ro/rollo-x1038/package.nix:
```diff
@@ -0,0 +1,30 @@
+  lib,
+}:
+
+stdenv.mkDerivation {
```
```suggestion
stdenv.mkDerivation (finalAttrs: {
```
Some attrs will be reused

acid-bong commented on pkgs/by-name/ro/rollo-x1038/package.nix:
```diff
@@ -0,0 +1,30 @@
+    maintainers = with maintainers; [ ChocolateLoverRaj ];
+    license = licenses.gpl3;
+  };
+}
```
```suggestion
})
```

acid-bong commented on pkgs/by-name/ro/rollo-x1038/package.nix:
```diff
@@ -0,0 +1,30 @@
+  name = "rollo-x1038";
+  version = "1.8.4";
+  src = fetchTarball {
+    url = "https://rollo-main.b-cdn.net/driver-dl/linux/rollo-cups-driver-1.8.4.tar.gz";
```
like here
```suggestion
    url = "https://rollo-main.b-cdn.net/driver-dl/linux/rollo-cups-driver-${finalAttrs.version}.tar.gz";
```

acid-bong commented on pkgs/by-name/ro/rollo-x1038/package.nix:
```diff
@@ -0,0 +1,30 @@
+stdenv.mkDerivation {
+  name = "rollo-x1038";
+  version = "1.8.4";
+  src = fetchTarball {
```
Also prefer Nixpkgs' own fetchers
```suggestion
  src = fetchzip {
```

acid-bong commented on pkgs/by-name/ro/rollo-x1038/package.nix:
```diff
@@ -0,0 +1,30 @@
+  stdenv,
+  cups,
+  lib,
+}:
```
```suggestion
  fetchzip,
}:
```

acid-bong commented on pkgs/by-name/ro/rollo-x1038/package.nix:
```diff
@@ -0,0 +1,30 @@
+  version = "1.8.4";
+  src = fetchTarball {
+    url = "https://rollo-main.b-cdn.net/driver-dl/linux/rollo-cups-driver-1.8.4.tar.gz";
+    sha256 = "sha256:0x8dd6wj5n974k3vb3iir61dd46ag13zj7s9dhgv722cky7v57h4";
```
```suggestion
    hash = "";
```
1. `hash` is preferred over type-specific fields
2. you'll need to regenerate hash (hint: you `nix-build` the package, see the hash mismatch error and copy the second one into the file)

acid-bong commented on pkgs/by-name/ro/rollo-x1038/package.nix:
```diff
@@ -0,0 +1,30 @@
+  nativeBuildInputs = [
+    cups
+  ];
+  installPhase = ''
+    install -d -m 755 $out/lib/cups/filter
+    install -c -m 755 rastertorollo $out/lib/cups/filter
+    install -d -m 755 $out/share/cups/model
+    install -c -m 644 rollo-x1038.ppd $out/share/cups/model
+  '';
```
```suggestion
  installPhase = ''
    runHook preInstall

    install -Dm755 rastertorollo -t $out/lib/cups/filter
    install -Dm644 rollo-x1038.ppd -t $out/share/cups/model

    runHook postInstall
  '';
```
1. Phases must remain adjustable (hence `runHook`)
2. These commands can be trimmed to just two

acid-bong commented on pkgs/by-name/ro/rollo-x1038/package.nix:
```diff
@@ -0,0 +1,30 @@
+  '';
+  meta = with lib; {
+    description = "CUPS Linux drivers for the Rollo X1038 thermal label printer";
+    downloadPage = "https://www.rollo.com/driver-linux/";
```
`homepage` also should be present

acid-bong commented on pkgs/by-name/ro/rollo-x1038/package.nix:
```diff
@@ -0,0 +1,30 @@
+    downloadPage = "https://www.rollo.com/driver-linux/";
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ ChocolateLoverRaj ];
+    license = licenses.gpl3;
```
`licenses.gpl#` (`#` is 2 or 3) are deprecated, use `gpl#Only` or `gpl#Plus` (`-or-later`). Judging by the lack of `or any later version` in license headers, it's `gpl3Only`

langsjo commented on pkgs/by-name/ro/rollo-x1038/package.nix:
```diff
@@ -0,0 +1,30 @@
+}:
+
+stdenv.mkDerivation {
+  name = "rollo-x1038";
```
```suggestion
  pname = "rollo-x1038";
```

langsjo commented on pkgs/by-name/ro/rollo-x1038/package.nix:
```diff
@@ -0,0 +1,30 @@
+    install -d -m 755 $out/share/cups/model
+    install -c -m 644 rollo-x1038.ppd $out/share/cups/model
+  '';
+  meta = with lib; {
+    description = "CUPS Linux drivers for the Rollo X1038 thermal label printer";
+    downloadPage = "https://www.rollo.com/driver-linux/";
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ ChocolateLoverRaj ];
+    license = licenses.gpl3;
+  };
```
```suggestion
  meta = {
    description = "CUPS Linux drivers for the Rollo X1038 thermal label printer";
    downloadPage = "https://www.rollo.com/driver-linux/";
    platforms = lib.platforms.linux;
    maintainers = with lib.maintainers; [ ChocolateLoverRaj ];
    license = lib.licenses.gpl3Only;
  };
```

langsjo reviewed (approved):
LGTM, just squash your commits into one commit with the proper commit message

acid-bong commented on pkgs/by-name/ro/rollo-x1038/package.nix:
```diff
@@ -0,0 +1,34 @@
+    url = "https://rollo-main.b-cdn.net/driver-dl/linux/rollo-cups-driver-${finalAttrs.version}.tar.gz";
+    hash = "sha256-BJ6yj59MiLMfbEkf+Ud4ypDWgskxjrXHJCfZIrlpDXU=";
+  };
+  nativeBuildInputs = [
+    cups
+  ];
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm755 rastertorollo -t $out/lib/cups/filter
+    install -Dm644 rollo-x1038.ppd -t $out/share/cups/model
+
+    runHook postInstall
+  '';
```
Ight, I found a solution in [another similar driver package](https://github.com/NixOS/nixpkgs/blob/eb35e32444d75a72c989be2324a4f4f706921515/pkgs/by-name/li/libppd/package.nix#L44-L48):
```suggestion
  nativeBuildInputs = [
    pkg-config
    cups
  ];
  buildInputs = [
    cups
  ];
  makeFlags = [
    "CUPS_SERVERBIN=$(out)/lib/cups"
    "CUPS_DATADIR=$(out)/share/cups"
    "CUPS_SERVERROOT=$(out)/etc/cups"
  ];

```

acid-bong replied:
Now the original `make install` target works properly (initially it wanted to create new dirs inside `cups`, and `./configure` doesn't have an option for that)

ChocolateLoverRaj committed:
rollo-x1038: init at 1.8.4

Tested with a Rollo X1038 printer.

Update pkgs/by-name/ro/rollo-x1038/package.nix

Co-authored-by: Acid Bong <acidbong@tilde.club>

Update pkgs/by-name/ro/rollo-x1038/package.nix

Co-authored-by: Acid Bong <acidbong@tilde.club>

Update pkgs/by-name/ro/rollo-x1038/package.nix

Co-authored-by: Acid Bong <acidbong@tilde.club>

Update pkgs/by-name/ro/rollo-x1038/package.nix

Co-authored-by: Acid Bong <acidbong@tilde.club>

Update pkgs/by-name/ro/rollo-x1038/package.nix

Co-authored-by: Acid Bong <acidbong@tilde.club>

Update pkgs/by-name/ro/rollo-x1038/package.nix

Co-authored-by: Acid Bong <acidbong@tilde.club>

Update pkgs/by-name/ro/rollo-x1038/package.nix

Co-authored-by: Acid Bong <acidbong@tilde.club>

Update pkgs/by-name/ro/rollo-x1038/package.nix

Co-authored-by: langsjo <104687438+langsjo@users.noreply.github.com>

Update pkgs/by-name/ro/rollo-x1038/package.nix

Co-authored-by: langsjo <104687438+langsjo@users.noreply.github.com>

Update hash and add homepage link

Update pkgs/by-name/ro/rollo-x1038/package.nix

Co-authored-by: Acid Bong <acidbong@tilde.club>

marcin-serwin commented on pkgs/by-name/ro/rollo-x1038/package.nix:
```diff
@@ -0,0 +1,36 @@
+  makeFlags = [
+    "CUPS_SERVERBIN=$(out)/lib/cups"
+    "CUPS_DATADIR=$(out)/share/cups"
+    "CUPS_SERVERROOT=$(out)/etc/cups"
```
This seems unused
```suggestion
```

ChocolateLoverRaj replied:
Idk I just copied the make flags from the script.

marcin-serwin reviewed (approved):
I don't have the printer so I can't test it but the recipe and and its output LGTM.

ChocolateLoverRaj commented:
I actually don't have this printer anymore because I returned it and got a Munbyn printer (#431890) instead. Btw I would recommend that one over this Rollo printer.

marcin-serwin commented:
> I actually don't have this printer anymore because I returned it

Arw you still interested in packaging and maintaining this driver in nixpkgs then? If not then please close the PR.

ChocolateLoverRaj commented:
I already tested that this works, but I don't want to maintain future updates for it because I can't test them. So if I should close this then I will.
